Xamarin/MvvmSeed/MvvmSeed/MvvmSeed.Android/FodyWeavers.xml

```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Weavers>
+  <RealmWeaver/>
```

Comment:
(Damn, I knew someone would ask about it..  😄 )

The complete answer is long.. but this is the configuration file for Fody, which is a tool to perform IL weaving (add extra code as part of the build process).

The common usage for IL weaving is to avoid boilerplate code..  so, for example, all the `INotifyPropertyChanged` boilerplate code to notify property changes and event definitions could be added automatically as part of the build process, as long as you can identity which properties need notification. That will save considerable time for developers and will keep the code clean.

The weaver for Realm looks something like this:

```
private void ReplaceGetter(PropertyDefinition prop, string columnName, MethodReference getValueReference)
    {
        // A synthesized property getter looks like this:
        //   0: ldarg.0
        //   1: ldfld <backingField>
        //   2: ret
        // We want to change it so it looks like this:
        //   0: ldarg.0
        //   1: call Realms.RealmObject.get_IsManaged
        //   2: brfalse.s 7
        //   3: ldarg.0
        //   4: ldstr <columnName>
        //   5: call Realms.RealmObject.GetValue<T>
        //   6: ret
        //   7: ldarg.0
        //   8: ldfld <backingField>
        //   9: ret
        // This is roughly equivalent to:
        //   if (!base.IsManaged) return this.<backingField>;
        //   return base.GetValue<T>(<columnName>);

        var start = prop.GetMethod.Body.Instructions.First();
        var il = prop.GetMethod.Body.GetILProcessor();

        il.InsertBefore(start, il.Create(OpCodes.Ldarg_0)); // this for call
        il.InsertBefore(start, il.Create(OpCodes.Call, _references.RealmObject_get_IsManaged));
        il.InsertBefore(start, il.Create(OpCodes.Brfalse_S, start));
        il.InsertBefore(start, il.Create(OpCodes.Ldarg_0)); // this for call
        il.InsertBefore(start, il.Create(OpCodes.Ldstr, columnName)); // [stack = this | name ]
        il.InsertBefore(start, il.Create(OpCodes.Call, getValueReference));
        il.InsertBefore(start, il.Create(OpCodes.Ret));

        Debug.Write("[get] ");
    }
```

At the same time, it performs some 'static analysis'. for example, if you declare a `DateTime` property that should be stored on the database, it will fail, since Realm used `DateTimeOffset`, which is more appropriate to handle dates on a db.

So, Fody is a tool for IL weaving, and on this file I'm configuring Realm weaver to do its 'magic' on the build process. This file can contain multiple weavers.. 

If this is 'clean' or not, it's another discussion.. but it is how Realm works on .NET.